Backend/auction-app/src/main/java/com/example/auctionapp/repository/BaseRepository.java

```diff
@@ -0,0 +1,12 @@
+package com.example.auctionapp.repository;
+
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class BaseRepository<T>
+        extends AbstractJpaDao<T> implements IBaseRepository<T>
```

Comment:
This is redundant, we can rename `AbstractJpaDao` to `BaseRepository` and make it implement `IBaseRepository<T>`